artiq/protocols/sync_struct.py

```diff
@@ -154,6 +155,8 @@ def __init__(self, notifier_name, target_builder, notify_cb=None, disconnect_cb=
 
                 for notify_cb in self.notify_cbs:
                     notify_cb(mod)
+        except (ConnectionError, asyncio.CancelledError):
```

Comment:
``Subscriber.close`` is already catching ``asyncio.CancelledError``. The logic is that ``wait_for`` will re-raise the exception that terminated the task. Why catch it again here?

Reply:
Okay, yes, I'll remove the CancelledError. Looking at it again, that was a work-around for a quamash bug that stopped exceptions propagating.